nodebuilder/tests/tastora/framework.go

```diff
@@ -0,0 +1,417 @@
+
+	f.t.Logf("Successfully funded node account: %s", nodeAddr.String())
+
+	return nodeAccAddr
+}
+
+// queryBalance fetches the balance of a given address.
+func (f *Framework) queryBalance(ctx context.Context, addr string) sdk.Coin {
+	grpcAddr := f.celestia.GetGRPCAddress()
+	grpcConn, err := grpc.NewClient(grpcAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(f.t, err, "failed to connect to gRPC")
+
+	bankClient := banktypes.NewQueryClient(grpcConn)
+	req := &banktypes.QueryBalanceRequest{
+		Address: addr,
+		Denom:   "utia",
+	}
+
+	res, err := bankClient.Balance(ctx, req)
+	require.NoError(f.t, err, "failed to query balance")
+	return *res.Balance
+}
+
+// createDockerProvider initializes the Docker provider for creating chains and nodes.
+func (f *Framework) createDockerProvider(cfg *Config) tastoratypes.Provider {
+	numValidators := cfg.NumValidators
+	numFullNodes := cfg.NumFullNodes
+
+	enc := testutil.MakeTestEncodingConfig(app.ModuleEncodingRegisters...)
+
+	dockerCfg := tastoradockertypes.Config{
+		Logger:          f.logger,
+		DockerClient:    f.client,
+		DockerNetworkID: f.network,
+		ChainConfig: &tastoradockertypes.ChainConfig{
+			ConfigFileOverrides: map[string]any{
+				"config/app.toml":    appOverrides(),
+				"config/config.toml": configOverrides(),
+			},
+			Type:          "cosmos",
+			Name:          "celestia",
+			Version:       getCelestiaTag(),
+			NumValidators: &numValidators,
+			NumFullNodes:  &numFullNodes,
+			ChainID:       testChainID,
+			Images: []tastoradockertypes.DockerImage{
+				{
+					Repository: celestiaAppImage,
+					Version:    getCelestiaTag(),
+					UIDGID:     "10001:10001",
+				},
+			},
+			Bin:            "celestia-appd",
+			Bech32Prefix:   "celestia",
+			Denom:          "utia",
+			CoinType:       "118",
+			GasPrices:      "0.025utia",
+			GasAdjustment:  1.3,
+			EncodingConfig: &enc,
+			AdditionalStartArgs: []string{
+				"--force-no-bbr",
+				"--grpc.enable",
+				"--grpc.address", "0.0.0.0:9090",
+				"--rpc.grpc_laddr", "tcp://0.0.0.0:9098",
+				"--timeout-commit", "1s",
+			},
+		},
+		DataAvailabilityNetworkConfig: &tastoradockertypes.DataAvailabilityNetworkConfig{
+			FullNodeCount:   cfg.FullNodeCount,
+			BridgeNodeCount: cfg.BridgeNodeCount,
+			LightNodeCount:  cfg.LightNodeCount,
+			Image: tastoradockertypes.DockerImage{
+				Repository: getNodeImage(),
+				Version:    getNodeTag(),
+				UIDGID:     "10001:10001",
+			},
+		},
+	}
+	return tastoradockertypes.NewProvider(dockerCfg, f.t)
+}
+
+// createAndStartCelestiaChain initializes and starts the Celestia chain.
+func (f *Framework) createAndStartCelestiaChain(ctx context.Context) tastoratypes.Chain {
+	celestia, err := f.provider.GetChain(ctx)
+	require.NoError(f.t, err, "failed to get chain")
+
+	err = celestia.Start(ctx)
+	require.NoError(f.t, err)
+
+	// verify the chain is producing blocks
+	require.NoError(f.t, wait.ForBlocks(ctx, 2, celestia))
+	return celestia
+}
```

Comment:
based on the work in the refactor I think just this function will need to change (not now), it will end up building the chain a different way, this is great that it's scoped to a small function so the impact should be minimal 👍